packages/vhd-lib/Vhd/VhdAbstract.integ.js

```diff
@@ -35,237 +24,250 @@ const streamToBuffer = stream => {
-    ]
-
-    for (const { targetPath, targetContent } of combinations) {
-      await testOneCombination({ targetPath, targetContent })
-    }
+let tempDir
+
+describe('VhdAbstract', async () => {
+  beforeEach(async () => {
+    tempDir = await pFromCallback(cb => tmp.dir(cb))
   })
-})
 
-test('alias must have *.alias.vhd extension', async () => {
-  await Disposable.use(async function* () {
-    const handler = yield getSyncedHandler({ url: 'file:///' })
-    const aliasPath = `${tempDir}/invalidalias.vhd`
-    const targetPath = `${tempDir}/targets.vhd`
-    expect(async () => await VhdAbstract.createAlias(handler, aliasPath, targetPath)).rejects.toThrow()
+  afterEach(async () => {
+    await rimraf(tempDir)
+  })
 
-    expect(await fs.exists(aliasPath)).toEqual(false)
+  it('Creates an alias', async () => {
+    await Disposable.use(async function* () {
+      const handler = yield getSyncedHandler({ url: 'file://' + tempDir })
+      const aliasPath = `alias/alias.alias.vhd`
+      const aliasFsPath = `${tempDir}/${aliasPath}`
+      await fs.mkdirp(`${tempDir}/alias`)
+
+      const testOneCombination = async ({ targetPath, targetContent }) => {
+        await VhdAbstract.createAlias(handler, aliasPath, targetPath)
+        // alias file is created
+        assert.equal(await fs.exists(aliasFsPath), true)
+        // content is the target path relative to the alias location
+        const content = await fs.readFile(aliasFsPath, 'utf-8')
+        assert.equal(content, targetContent)
+        // create alias fails if alias already exists, remove it before next loop step
+        await fs.unlink(aliasFsPath)
+      }
+
+      const combinations = [
+        { targetPath: `targets.vhd`, targetContent: `../targets.vhd` },
+        { targetPath: `alias/targets.vhd`, targetContent: `targets.vhd` },
+        { targetPath: `alias/sub/targets.vhd`, targetContent: `sub/targets.vhd` },
+        { targetPath: `sibling/targets.vhd`, targetContent: `../sibling/targets.vhd` },
+      ]
+
+      for (const { targetPath, targetContent } of combinations) {
+        await testOneCombination({ targetPath, targetContent })
+      }
+    })
   })
-})
 
-test('alias must not be chained', async () => {
-  await Disposable.use(async function* () {
-    const handler = yield getSyncedHandler({ url: 'file:///' })
-    const aliasPath = `${tempDir}/valid.alias.vhd`
-    const targetPath = `${tempDir}/an.other.valid.alias.vhd`
-    expect(async () => await VhdAbstract.createAlias(handler, aliasPath, targetPath)).rejects.toThrow()
-    expect(await fs.exists(aliasPath)).toEqual(false)
+  it('has *.alias.vhd extension', async () => {
+    await Disposable.use(async function* () {
+      const handler = yield getSyncedHandler({ url: 'file:///' })
+      const aliasPath = `${tempDir}/invalidalias.vhd`
+      const targetPath = `${tempDir}/targets.vhd`
+      assert.rejects(async () => await VhdAbstract.createAlias(handler, aliasPath, targetPath))
```

Comment:
`assert.rejects()` returns a promise which MUST be awaited.

Reply:
You're right!

Reply:
Either to fix in a new commit or in the correct commit! (I just rebased the branch to fix some changes which were not in the correct commit).